### PR TITLE
Add xmldoc and rename methods in `IPositionSnapProvider` for legibility

### PR DESCRIPTION
**Pull request by peppy**
I'm going through the snapping logic in general and tidying things up to a point I can understand the flow again. This is the first of (probably) a few PRs to get things back in line. Keeping them bite-sized to make sure they can be merged and reviewed easily.

### CODE REVIEW

File: osu.Game/Rulesets/Edit/IPositionSnapProvider.cs [hunk 1]
```diff
@@ -5,23 +5,27 @@
...
-        /// This call should be equivalent to running <see cref="SnapScreenSpacePositionToValidPosition"/> with any additional logic that can be performed without the time immutability restriction.
+        /// This call should be equivalent to running <see cref="FindSnappedPosition"/> with any additional logic that can be performed without the time immutability restriction.
         /// </remarks>
         /// <param name="screenSpacePosition">The screen-space position to be snapped.</param>
         /// <returns>The time and position post-snapping.</returns>
-        SnapResult SnapScreenSpacePositionToValidTime(Vector2 screenSpacePosition);
+        SnapResult FindSnappedPositionAndTime(Vector2 screenSpacePosition);
 
         /// <summary>
         /// Given a position, find a value position snap, restricting time to its input value.
```
Comment (frenzibyte) [inline osu.Game/Rulesets/Edit/IPositionSnapProvider.cs:25]: I understand this part existed before this PR, but since we're cleaning things here...
```suggestion
        /// Given a position, find a valid position snap, restricting time to its input value.
```

Also I'm not sure what the point of this `SnapScreenSpacePositionToValidPosition`/`FindSnappedPosition` method is and why it returns `SnapResult` in the first place, feels like it could indeed be simplified to a boolean (or just not exist in the first place and not pay attention to the resultant time value..?)

At least it makes much more sense now with the rename.

**Review by frenzibyte — COMMENTED** [review]
Reads pretty well otherwise.

File: osu.Game/Rulesets/Edit/IPositionSnapProvider.cs (hunk 1 above)
Comment (bdach) [inline osu.Game/Rulesets/Edit/IPositionSnapProvider.cs:25]: While we're here adjusting xmldocs, am I the only one that finds this one illegible? "value position snap" seems to be a typo, but I'd generally reword this to something like

```suggestion
        /// Given a position, find a valid position snap, without changing the time value.
```

Comment (bdach) [reply osu.Game/Rulesets/Edit/IPositionSnapProvider.cs]: Let's keep return value changes out of this pull please. Maybe separately if it's deemed better.

Comment (frenzibyte) [reply osu.Game/Rulesets/Edit/IPositionSnapProvider.cs]: Definitely for a separate PR, just bringing it here for note.

**Review by bdach — APPROVED** [review]
